source/core/coverage/ut_block_helper.pkb

```diff
@@ -0,0 +1,89 @@
+create or replace package body ut_block_helper is
+  /*
+  utPLSQL - Version 3
+  Copyright 2016 - 2017 utPLSQL Project
+
+  Licensed under the Apache License, Version 2.0 (the "License"):
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+  */
+
+  type t_proftab_row is record (
+      line  binary_integer,
+      calls number(38,0)
+    );
+    
+  type t_proftab_rows is table of t_proftab_row;
+
+  type t_block_row is record(
```

Comment:
I would transform the not-so-good Oracle results into:

- start_line  binary_integer
- end_line  binary_integer - (`lead(line,1) over(order by line)` from coverage data - null if no next line)
- blocks     binary_integer
- covered_blocks binary_integer

I assume that there is no situation where we have more than 1 block spanning across more than one line.
That is:
only when start_line = end_line we can get blocks>1

Reply:
I’m afraid it might be possible for complex multi line `if` Statement

Reply:
Even when simple if and bad beutifier we would have one block in single line.

Reply:
Not sure if thats really do the trick as we cant really tell more details about column line of the block.Also graphical represenation on the html report or in fact even xml would struggle to present these data.